core/src/main/java/hudson/PluginManager.java

```diff
@@ -219,6 +230,14 @@ public Api getApi() {
     }
 
     /**
+     * If non-null, the base directory for all exploded .hpi/.jpi plugins.
+     * @return the base directory for all exploded .hpi/.jpi plugins or {@code null} to leave this up to the strategy.
+     */
+    public File getWorkDir() {
```

Comment:
So make the default for `rootDir` `baseDir/plugins` ?  Any other PluginStrategy will be unaffected until they update to the new API?

FWIW the correct annotations are the deprecated edu.cs...findbugs ones :-P